src/ch/imedias/rsccfx/model/Rscc.java

```diff
@@ -15,22 +24,79 @@
    * Important: Make sure to NOT include a / in the beginning or the end.
    */
   private static final String DOCKER_FOLDER_NAME = "docker-build_p2p";
-  private final String pathToResourceDocker;
-  private final String scriptShell;
-  private final ProcessExecutor processExecutor;
+  private static final String RSCC_FOLDER_NAME = ".rscc";
+  private SystemCommander systemCommander;
+  private String pathToResourceDocker;
   private StringProperty key = new SimpleStringProperty();
   private String keyServerIp;
   private String keyServerHttpPort;
 
   /**
    * Initializes the Rscc model class.
    */
-  public Rscc(ProcessExecutor processExecutor) {
-    this.processExecutor = processExecutor;
-    pathToResourceDocker =
-        getClass().getClassLoader().getResource(DOCKER_FOLDER_NAME)
-            .getFile().toString().replaceFirst("file:", "");
-    scriptShell = "bash" + " " + pathToResourceDocker + "/";
+  public Rscc(SystemCommander systemCommander) {
+    this.systemCommander = systemCommander;
+    defineResourcePath();
+  }
+
+  /**
+   * Sets resourcepath - dependet if application runs as JAR or in IDE.
+   */
+  private void defineResourcePath() {
+    String userhome = System.getProperty("user.home");
+    URL location = this.getClass().getProtectionDomain().getCodeSource().getLocation();
+    File f = new File(location.getFile());
+    if (f.isDirectory()) {
+      pathToResourceDocker =
+              getClass().getClassLoader().getResource(DOCKER_FOLDER_NAME)
+                      .getFile().toString().replaceFirst("file:", "");
+
+    } else {
+      pathToResourceDocker = userhome + "/" + RSCC_FOLDER_NAME + "/" + DOCKER_FOLDER_NAME;
+      extractJarContents(location, userhome + "/" + RSCC_FOLDER_NAME, DOCKER_FOLDER_NAME);
+    }
+  }
+
+  /**
+   * Extract files from running JAR to folder.
+   */
+  private void extractJarContents(URL sourceLocation, String destinationDirectory, String filter) {
+    JarFile jarfile = null;
+    try {
+      jarfile = new JarFile(new File(sourceLocation.getFile()));
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+
+    Enumeration<JarEntry> enu = jarfile.entries();
```

Comment:
It creates an Enumerator: http://download.java.net/java/jdk9/docs/api/java/util/Enumeration.html of the content of the JAR-File